### PR TITLE
refactor: replaced deprecated ioutil calls with os equivalents

### PR DESCRIPTION
**Pull request by ardevd**

### CODE REVIEW

File: basic_client_test.go [hunk 1]
```diff
@@ -46,21 +45,21 @@ XhkpT5dliEGFLNe6OOgeWTU1JpEXfCud/GImtNMHQi4EDWQfvWuCNGhOoQ==
 
 	// Now let's write the data to a file to make sure parseTLSAndMacaroon
 	// parses that properly as well.
-	tempDirPath, err := ioutil.TempDir("", ".testCreds")
+	tempDirPath, err := os.MkdirTemp("", ".testCreds")
```
Comment (guggero) [inline basic_client_test.go:48]: Can even use `t.TempDir()`, then the `defer os.RemoveAll(tempDirPath)` is done automatically.

**Review by guggero — APPROVED** [review]
LGTM, thanks!

**Review by bhandras — APPROVED** [review]
LGTM, thanks! 🌮

File: basic_client_test.go (hunk 1 above)
Comment (ardevd) [reply basic_client_test.go]: Nice! Agree!

**Comment by guggero** [comment]
Linter is not happy because of the extra space.

**Comment by ardevd** [comment]
Fixed!

**Pull request merged**